Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -489,26 +508,33 @@ public ParserState GetModuleState(QualifiedModuleName module)
         private ParserState _status;
         public ParserState Status
         {
-            get { return _status; }
-            private set
+            get => _status;
+            private set => SetStatusWithCancellation(value, CancellationToken.None);
+        }
+
+        private void SetStatusWithCancellation(ParserState value, CancellationToken token)
+        {
+            if (_status != value)
             {
-                if (_status != value)
-                {
-                    _status = value;
-                    OnStateChanged(this, _status);
-                }
+                _status = value;
+                OnStateChanged(this, token, _status);
             }
         }
 
         public void SetStatusAndFireStateChanged(object requestor, ParserState status)
+        {
+            SetStatusAndFireStateChanged(requestor, status, CancellationToken.None);
```

Comment:
I think that similarly, a token ought to be requested or provided here as well.

Reply:
Currently, these overloads are not used. Thay are just making the token an optional parameter. (It cannot be made a real optional one because `CancellationToken` is a struct and `CancellationToken.None` not a compile time constant.)

Should I remove the overloads?

Reply:
We want to reduce the possibilities of incorrectly using the internal API as much as possible. Especially since we want to **enforce** cancellability, it's counterproductive to have these overloads. 

I'm strongly in favor of removing them